flow/project.py

```diff
@@ -940,19 +939,9 @@ def _get_submission_directives(self, default_directives, job, parallel=False):
         """
         directives = dict(ngpu=0, nranks=0, omp_num_threads=0, np=0)
 
-        def get_directive(directives, key, default, job):
-            d = directives.get(key, default)
-            return d(job) if callable(d) else d
-
         for name in self.operations:
             # get directives for operation
-            op_dir = self._resolve_directives(name, default_directives)
-            # if operations are callable handle appropriately with job
-            op_dir.update(dict(
-                ngpu=get_directive(op_dir, 'ngpu', 0, job),
-                nranks=get_directive(op_dir, 'nranks', 0, job),
-                omp_num_threads=get_directive(op_dir, 'omp_num_threads', 0, job)))
-
+            op_dir = self._resolve_directives(name, default_directives, job)
             # Find the correct number of processors for operation
             np = op_dir.get('np', max(op_dir['nranks'], 1) * max(op_dir['omp_num_threads'], 1))
```

Comment:
This logic for `np` is already in `_resolve_directives`, so we can safely remove it.